### DISCUSSION TITLE
Text-to-Speech

### DISCUSSION THREAD

**Discussion by tautologias345** [godotengine/godot-proposals#6387] [GUI]
The make a text-to-speech, the developer should choose the gender (male, female), the human age (from 2 to 125) and the voice style:

1. neutral voice
2. scared voice
3. sleepy voice
4. furious voice
5. crying voice
6. upset voice
7. laughing voice
8. deep voice
9. high-pitched voice
10. scream voice
11. whisper voice
12. joy voice
13. achievement voice
14. failure voice
15. hurried voice
16. doing a please voice
17. many other voice styles

**Comment by Zireael07** [comment]
TTS is not something that can be used to achieve what you're looking for. Haven't seen a single TTS capable of conveying emotions

**Comment by dukemagus** [comment]
Not how it works. if you have a TTS library and all the necessary assets, you can create an EXTENSION to the engine with that. but by itself it shouldn't be a core feature

You can't just call a generic TTS api because each platform has its own TTS solution (microsoft sam, google tts for android, whatever apple uses. etc.)
